### ISSUE TITLE
retouch wavelet decomposition & colorpicker in fill color mode: color box doesn't show picked color

### ISSUE THREAD

**Issue by MStraeten** [darktable-org/darktable#8894]
color of the picker isn't identical to the color the picker is over
![image](https://user-images.githubusercontent.com/39386816/117534040-0bce1880-aff0-11eb-864c-0f2ef2ce01be.png)

at least painting around with the color in fill mode uses the color the picker was over, not the color displayed in the color box
 
* darktable version : 3.5.0+2120~g3c608f9a95-dirty
* OS                :  OSx

**Comment by MStraeten** [comment]
still reproducible with current master

**Comment by MStraeten** [comment]
keeps being reproducible with darktable 4.1.0+1033~g688a4ae04

**Comment by MStraeten** [comment]
keeps being reproducible with 4.3.0+1939~g8eb9ad90ba

**Comment by ralfbrown** [comment]
If I understand your issue correctly, you want to be able to pick a color from the residual layer of the wavelet decomposition?

As far as the "strange encoding", the residual layer is literally what is left after subtracting off all of the wavelet layers, which makes each pixel basically the average color of a square 2**layers pixels on a side. It's still in the pixelpipe's working color space.

**Comment by MStraeten** [comment]
yes, i enable the display of the residual layer and pick a color - but the picked "fill color" doesn't match the displayed color.

For me it doesn't make sense to process the residual layer preview with the later modules in the pixel pipe - what you see isn't what you get

**Comment by vredesbyyrd** [comment]
> For me it doesn't make sense to process the residual layer preview with the later modules in the pixel pipe - what you see isn't what you get

Agreed. I ran into this today. I just started learning the retouch -> fill tool and the picked "fill color" not matching the displayed color has thrown me off, eg, picking very pale skin tone the displayed color is black.
